drgn/helpers/linux/kernfs.py

```diff
@@ -61,28 +63,44 @@ def kernfs_path(kn: Object) -> bytes:
     return b"/".join(names)
 
 
-def kernfs_walk(parent: Object, path: Path) -> Object:
+def kernfs_walk(parent: Object, path: Path, follow_symlinks: bool = False) -> Object:
     """
     Find the kernfs node with the given path from the given parent kernfs node.
 
     :param parent: ``struct kernfs_node *``
     :param path: Path name.
+    :param follow_symlinks: If follow_symlinks is ``False``, and the
+        last component of a path is a symlink, the function will
+        return ``struct kernfs_node *`` of the symbolic link.
```

Comment:
Why only the last component?

I guess I don't know well enough whether there are cases where symlinks would be in the middle of a real kernfs path. So maybe it won't make a difference. It just seems reasonable to support it within the path too.

Reply:
That probably came from my description in #443. My intention was that internal symlinks are always followed, and the last one is controlled by the parameter, just like `O_NOFOLLOW` and the `os` module. I haven't read the code to see if there was a misunderstanding.

Reply:
Ok, back from vacation and read the code now. Yes, this needs to be changed to always follow symlinks that are not the last component.